cns/networkcontainers/networkcontainers.go

```diff
@@ -57,3 +91,105 @@ func (cn *NetworkContainers) Delete(networkContainerID string) error {
 	log.Printf("[Azure CNS] NetworkContainers.Delete finished.")
 	return err
 }
+
+// This function gets the flattened network configuration (compliant with azure cni) in byte array format
+func getNetworkConfig(configFilePath string) ([]byte, error) {
+	content, err := ioutil.ReadFile(configFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	var configMap map[string]interface{}
+	if err = json.Unmarshal(content, &configMap); err != nil {
+		log.Printf("[Azure CNS] Failed to unmarshal network configuration with error %v", err)
+		return nil, err
+	}
+
+	// Get the plugins section
+	var flatNetConfigMap map[string]interface{}
+	if pluginsSection, ok := configMap[pluginsStr]; ok {
+		flatNetConfigMap = pluginsSection.([]interface{})[0].(map[string]interface{})
```

Comment:
can't pluginSection be empty array? do we need to check if len(pluginsSection) >0 here..? if you feel its not needed then mark this as resolved. i'm not stressing it